src/pages/Register/Register.tsx

```diff
@@ -0,0 +1,198 @@
+import React from "react";
+import { Grid, TextField, Button, Typography, Link } from "@mui/material";
+import { useFormik } from "formik";
+import * as yup from "yup";
+import "./Register.css";
+import logo from "../../assets/images/logo.png";
+
+// Validation schema
+const validationSchema = yup.object({
+  firstName: yup.string().required("First Name is required"),
+  lastName: yup.string().required("Last Name is required"),
+  username: yup.string().required("Username is required"),
+  password: yup
+    .string()
+    .required("Password is required")
+    .min(8, "Password must be at least 8 characters"),
+  confirmPassword: yup
+    .string()
+    .required("Confirm Password is required")
+    .oneOf([yup.ref("password")], "Passwords must match"),
+});
+
+const Register: React.FC = () => {
+  const formik = useFormik({
+    initialValues: {
+      firstName: "",
+      lastName: "",
+      username: "",
+      password: "",
+      confirmPassword: "",
+    },
+    validationSchema: validationSchema,
+    onSubmit: (values: any) => {
+      console.log(values);
+    },
+  });
+
+  return (
+    <Grid
+      container
+      className="register-container"
+      alignItems="center"
+      justifyContent="center"
+    >
+      <Grid item xs={12} md={4} className="register-left">
+        <img src={logo} alt="Logo" className="logo-image" />
+        <div className="logo-info">
+          <Typography
+            variant="h2"
+            className="logo-name"
+            sx={{
+              backgroundImage:
+                "linear-gradient(98.63deg, #f9a225 0%, #f95f35 100%)",
+              fontWeight: 800,
+              color: "transparent",
+              backgroundClip: "text",
+              WebkitBackgroundClip: "text",
+              WebkitTextFillColor: "transparent",
+            }}
+          >
+            ZKC Media
+          </Typography>
+          <Typography variant="body1" className="tagline">
+            Explore the ideas throughout the world
+          </Typography>
+        </div>
+      </Grid>
+      <Grid
+        item
+        xs={12}
+        md={4.5}
+        className="register-form-container"
+        sx={{ borderRadius: 5 }}
+      >
+        <Typography
+          variant="h4"
+          className="register-title"
+          sx={{ fontWeight: 600 }}
+        >
+          Sign up
+        </Typography>
+        <form onSubmit={formik.handleSubmit}>
+          <Grid container spacing={2}>
+            <Grid item xs={6}>
+              <TextField
+                fullWidth
+                label="First Name"
+                margin="normal"
+                variant="filled"
+                id="firstName"
+                name="firstName"
+                value={formik.values.firstName}
+                onChange={formik.handleChange}
+                onBlur={formik.handleBlur}
+                error={
+                  formik.touched.firstName && Boolean(formik.errors.firstName)
```

Comment:
can be written 
!!(formik.errors?.firstName)

Reply:
when I write it "!!(formik.errors?.firstName)" then all fields return error when I toch one field , like this screenShot
Should I use the same phrase or keep it as it was?
![image](https://github.com/batool2azzam/Social-Media-App/assets/122049115/62a57f2b-fc37-415d-8f8a-2258890082a2)

Reply:
it should be 
{ formik.touched.firstName && !!(formik.errors.firstName)}

!! should convert any value to its corresponding boolean value